### DISCUSSION TITLE
Feature request: Branch rules to optionally allow regex or fnmatch

### DISCUSSION THREAD

**Discussion by RichardTMiles** [community/community#53523] [Repositories]
### Select Topic Area

Product Feedback

### Body

Firstly, thanks for taking the time to read this!

There is a real need to allow regex in Branch rules. I advocate that it is more widely taught in universities, boot camps, etc. It also dramatically simplifies complex naming convention rules. I have attached some top hits on Google for the community's pain. Yes, there are workarounds// learning curves // pre-commit hooks to solve the issue; but the community value for this feature would be well worth it. 

Best,
@RichardTMiles

https://stackoverflow.com/questions/55053460/github-branch-name-pattern-negation
https://stackoverflow.com/questions/53135414/how-to-apply-one-github-branch-rule-to-multiple-branches
https://stackoverflow.com/questions/68577301/github-branch-protection-rule-pattern-for-set-branch-names

**Comment by patrick-knight** [comment]
Responded over here: https://github.com/orgs/community/discussions/52652#discussioncomment-5759391
